corehq/apps/callcenter/indicator_sets.py

```diff
@@ -348,16 +348,21 @@ def add_form_data(self, indicator_config):
                 logger.debug('Adding legacy forms submitted stats: %s', indicator_config.date_ranges)
                 self._add_data(results, '{}{}'.format(LEGACY_FORMS_SUBMITTED, range_name.title()))
 
-            if self.domain in PER_DOMAIN_FORM_INDICATORS:
-                for custom in PER_DOMAIN_FORM_INDICATORS[self.domain]:
-                    self.add_custom_form_data(
-                        custom['slug'],
-                        range_name,
-                        custom['xmlns'],
-                        custom['type'],
-                        lower,
-                        upper
-                    )
+    def add_custom_data(self, indicator_config):
+        custom_domain_indicators = PER_DOMAIN_FORM_INDICATORS.get(self.domain, {})
+        for indicator in indicator_config:
+            custom = custom_domain_indicators.get(indicator.type)
+            if not custom:
+                continue  # domain does not have indicator of the specified name
+            lower, upper = self.date_ranges[indicator.date_range]
+            self.add_custom_form_data(
+                indicator.type,
+                indicator.date_range,
+                custom['xmlns'],
+                custom['type'],
+                lower,
+                upper
+            )
```

Comment:
instead of adding all the custom ones, only include ones that are in the `indicator_config`